js/init_and_modules_test.go

```diff
@@ -0,0 +1,106 @@
+package js_test
+
+import (
+	"context"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/loadimpact/k6/js"
+	"github.com/loadimpact/k6/js/common"
+	"github.com/loadimpact/k6/js/internal/modules"
+	"github.com/loadimpact/k6/lib"
+	"github.com/loadimpact/k6/lib/testutils"
+	"github.com/loadimpact/k6/loader"
+	"github.com/loadimpact/k6/stats"
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/guregu/null.v3"
+)
+
+type CheckModule struct {
+	t             testing.TB
+	initCtxCalled int
+	vuCtxCalled   int
+}
+
+func (cm *CheckModule) InitCtx(ctx context.Context) {
+	cm.initCtxCalled++
+	assert.NotNil(cm.t, common.GetRuntime(ctx))
+	assert.NotNil(cm.t, common.GetInitEnv(ctx))
+	assert.Nil(cm.t, lib.GetState(ctx))
+}
+
+func (cm *CheckModule) VuCtx(ctx context.Context) {
+	cm.vuCtxCalled++
+	assert.NotNil(cm.t, common.GetRuntime(ctx))
+	assert.Nil(cm.t, common.GetInitEnv(ctx))
+	assert.NotNil(cm.t, lib.GetState(ctx))
+}
+
+func TestNewJSRunnerWithCustomModule(t *testing.T) {
+	checkModule := &CheckModule{t: t}
+	modules.Register("k6/check", checkModule)
```

Comment:
We just have to remember to use different packages for registering new k6 modules that help with testing... Given the singleton nature of the module registry, we don't want "pollution"

Reply:
Yeah, that could be a problem. Should we add some kind of `Deregister` mechanism so that each test can clean up after itself? Since it's in `internal` we don't have to make it public and could only use it in tests.

Reply:
it shouldn't really matter if such tests are in their own package, but sure, an internal de-registering function can be added in the future